### PR TITLE
Issue/4075 failed upload error message

### LINKED ISSUE #4075: Failed image upload error message
When WordPress.com users reach the maximum capacity of their Media Library they can no longer upload images. The app doesn't explain why they can't upload images and only says "Failed". It would be great to let these users know that they either need to buy more space or free up some space in their Media Library first.

Here's what they see currently.

slack-imgs com

### PR DESCRIPTION
**Pull request by SergioEstevao**
Closes #4075 

Needs Review: @astralbodies

### CODE REVIEW

File: WordPress/Classes/Services/MediaService.m [hunk 1]
```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
...
+    NSError *newError = error;
+    if (error.domain == WordPressComApiErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case WordPressComApiErrorUploadFailed:{
+                errorMessage = NSLocalizedString(@"The app couldn't upload this media.", @"Message to show to user when media upload failed by unknow server error");
+                errorCode = WordPressComApiErrorUploadFailed;
+            } break;
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: We aren't explicit with styling of case blocks in our style guide but I want to say this isn't the norm across the app. You really don't need to local scope those two variable assignments.

```objc
case WordPressComApiErrorUploadFailed:
    errorMessage = NSLocalizedString(@"The app couldn't upload this media.", @"Message to show to user when media upload failed by unknow server error");
    errorCode = WordPressComApiErrorUploadFailed;
    break;
```

File: WordPress/Classes/Services/MediaService.m [hunk 2]
```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
                 failure:failureBlock];
 }
 
+- (NSError *)processMediaUploadError:(NSError *)error {
```
Comment (astralbodies) [inline WordPress/Classes/Services/MediaService.m]: It's a nitpick but maybe `translateMediaUploadError` - process don't really declare what it does

**Comment by SergioEstevao** [comment]
@astralbodies ready for another round of review.

**Comment by astralbodies** [comment]
Sounds good @SergioEstevao - getting a multiuser install working at the moment. :smile:

**Comment by astralbodies** [comment]
Awesome, lets :shipit:

File: WordPress/Classes/Services/MediaService.m [hunk 3]
```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
...
+    } else if (error.domain == WPXMLRPCFaultErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case 500:{
+                errorMessage = NSLocalizedString(@"Your site does not support this media file format.", @"Message to show to user when media upload failed because server doesn't support media type");
+                errorCode = WordPressComApiErrorUploadFailedInvalidFileType;
+            } break;
+            case 401:{
```
Comment (maxme) [inline WordPress/Classes/Services/MediaService.m]: 401 error can happen when the user doesn't have the rights to upload media (Subscriber role for instance).

File: WordPress/Classes/Services/MediaService.m [hunk 4]
```diff
@@ -190,6 +192,48 @@ - (void)uploadMedia:(Media *)media
...
+        }
+        userInfo[NSLocalizedDescriptionKey] = errorMessage;
+        newError = [[NSError alloc] initWithDomain:WordPressComApiErrorDomain code:errorCode userInfo:userInfo];
+    } else if (error.domain == WPXMLRPCFaultErrorDomain) {
+        NSString *errorMessage = [error localizedDescription];
+        NSInteger errorCode = error.code;
+        NSMutableDictionary *userInfo = [NSMutableDictionary dictionaryWithDictionary:error.userInfo];
+        switch (error.code) {
+            case 500:{
+                errorMessage = NSLocalizedString(@"Your site does not support this media file format.", @"Message to show to user when media upload failed because server doesn't support media type");
```
Comment (maxme) [inline WordPress/Classes/Services/MediaService.m]: Same kind of comment: 500 code means a lot of different things :/ (especially on wporg installs) I'm not sure we should use that error message here.

**Comment by SergioEstevao** [comment]
@maxme Just show the message that wp.org returns? we can do that, because from my tests it's a good string and proper translated...

**Comment by maxme** [comment]
> @maxme Just show the message that wp.org returns?

Yes, that's what I implemented in https://github.com/wordpress-mobile/WordPress-Android/pull/3770 - I think a too technical message is better than a wrong message (even if it happens in some cases only).

**Pull request merged**